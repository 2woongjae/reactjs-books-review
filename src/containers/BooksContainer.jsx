import React, { useCallback } from "react";
import { useSelector, useDispatch } from "react-redux";
import Books from "../components/Books";
import { requestBooksSaga } from "../redux/modules/books";

const BooksContainer = props => {
  const books = useSelector(state => state.books.books);
  const loading = useSelector(state => state.books.loading);
  const error = useSelector(state => state.books.error);

  const dispatch = useDispatch();

  const requestBooks = useCallback(() => {
    dispatch(requestBooksSaga());
  }, [dispatch]);

  return (
    <Books
      {...props}
      books={books}
      loading={loading}
      error={error}
      requestBooks={requestBooks}
    />
  );
};

export default BooksContainer;
