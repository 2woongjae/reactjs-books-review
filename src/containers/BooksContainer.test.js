import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import configureMockStore from "redux-mock-store";
import BooksContainer from "./BooksContainer";

Enzyme.configure({ adapter: new Adapter() });

describe("BooksContainer", () => {
  it("renders properly", () => {
    const mockStore = configureMockStore();

    const store = mockStore({
      books: [
        {
          bookId: 1,
          ownerId: "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
          title: "모던 자바스크립트 입문",
          message: "모던하군요"
        },
        {
          bookId: 2,
          ownerId: "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
          title: "책 Mock",
          message: "메세지 Mock"
        }
      ],
      loading: false,
      error: null
    });

    const component = mount(<BooksContainer store={store} />);
    expect(component).toMatchSnapshot();
  });
});
