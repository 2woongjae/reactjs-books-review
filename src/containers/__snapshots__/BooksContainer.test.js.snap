// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BooksContainer renders properly 1`] = `
<Connect(Books)
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Books
    books={
      Array [
        Object {
          "bookId": 1,
          "message": "모던하군요",
          "ownerId": "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
          "title": "모던 자바스크립트 입문",
        },
        Object {
          "bookId": 2,
          "message": "메세지 Mock",
          "ownerId": "7d26db27-168c-4c6a-bd9a-9e20677b60b8",
          "title": "책 Mock",
        },
      ]
    }
    error={null}
    loading={false}
    requestBooksPromise={[Function]}
    requestBooksSaga={[Function]}
    requestBooksThunk={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <div>
      <Book
        key="1"
        title="모던 자바스크립트 입문"
      >
        <div>
          title : 
          모던 자바스크립트 입문
        </div>
      </Book>
      <Book
        key="2"
        title="책 Mock"
      >
        <div>
          title : 
          책 Mock
        </div>
      </Book>
    </div>
  </Books>
</Connect(Books)>
`;
